app/main.py

```diff
@@ -1,34 +1,99 @@
+from typing import Dict, List, Tuple, Union
+
+
 class Deck:
-    def __init__(self, row, column, is_alive=True):
-        pass
+    def __init__(
+            self,
+            row: int,
+            column: int,
+            is_alive: bool = True,
+            is_hit: bool = False
+    ) -> None:
+        self.row = row
+        self.column = column
+        self.is_alive = is_alive
+        self.is_hit = is_hit
+
+    def fire(self) -> None:
+        self.is_alive = False
 
 
 class Ship:
-    def __init__(self, start, end, is_drowned=False):
-        # Create decks and save them to a list `self.decks`
-        pass
+    def __init__(
+            self,
+            start: Tuple[int, int],
+            end: Tuple[int, int],
+            is_drowned: bool = False
+    ) -> None:
+        self.start = start
+        self.end = end
+        self.is_drowned = is_drowned
+        self.decks = []
+        (start_row, start_column), (end_row, end_column) = start, end
+
+        for row in range(start_row, end_row + 1):
+            for column in range(start_column, end_column + 1):
+                deck = Deck(row, column)
+                self.decks.append(deck)
 
-    def get_deck(self, row, column):
-        # Find the corresponding deck in the list
-        pass
+    def get_deck(self, row: int, column: int) -> Union[Deck, None]:
+        for deck in self.decks:
+            if deck.row == row and deck.column == column:
+                return deck
 
-    def fire(self, row, column):
-        # Change the `is_alive` status of the deck
-        # And update the `is_drowned` value if it's needed
-        pass
+    def fire(self, row: int, column: int) -> None:
+        deck = self.get_deck(row, column)
+        if deck is not None:
+            deck.fire()
+            if not self.is_drowned and all(not d.is_alive for d in self.decks):
+                self.is_drowned = True
```

Comment:
You can use 
```suggestion
            self.is_drowned = (
                not self.is_drowned and all(not deck.is_alive for deck in self.decks)
            )
```
Also d is not good name